core/src/test/java/io/questdb/test/std/FdCacheCounterOverflowTest.java

```diff
@@ -0,0 +1,355 @@
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.test.std;
+
+import io.questdb.std.Files;
+import io.questdb.std.MemoryTag;
+import io.questdb.std.Misc;
+import io.questdb.std.Os;
+import io.questdb.std.Unsafe;
+import io.questdb.std.str.Path;
+import io.questdb.test.AbstractTest;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static io.questdb.test.tools.TestUtils.assertMemoryLeak;
+
+public class FdCacheCounterOverflowTest extends AbstractTest {
+    private static final byte CONTENT_A = 'A';
+    private static final byte CONTENT_B = 'B';
+    private static final int FILE_SIZE = 1024;
+    private Path testFileA;
+    private Path testFileB;
+
+    @Before
+    public void setUp() {
+        super.setUp();
+        testFileA = new Path();
+        testFileB = new Path();
+        setupTestFiles();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        Misc.free(testFileA);
+        Misc.free(testFileB);
+        super.tearDown();
+    }
+
+    @Test
+    public void testConcurrentOpenClose() throws Exception {
+        // Concurrent test that spawns a single writer that writes an append-only file, and multiple workers.
+        // It sets the FD counter near the overflow initially, and the reader threads actively open/read/close the file.
+
+        Files.setFDCacheCounter(Integer.MAX_VALUE - 100);
+
+        assertMemoryLeak(() -> {
+            final int numReaderThreads = 4;
+            final int testDurationMs = 5000;
+            final Thread[] readerThreads = new Thread[numReaderThreads];
+            final Throwable[] exceptions = new Throwable[numReaderThreads + 1];
+            final AtomicBoolean shouldStop = new AtomicBoolean(false);
+
+            // Writer thread - continuously appends to file A
+            Thread writerThread = new Thread(() -> {
+                byte[] appendData = new byte[16];
```

Comment:
nit: why don't we use mmap for writes too to keep the scenario closer to what we have in the code base?